ReactAndroid/src/main/java/com/facebook/react/bridge/ReadableNativeArray.java

```diff
@@ -46,4 +49,38 @@ protected ReadableNativeArray(HybridData hybridData) {
   public native ReadableNativeMap getMap(int index);
   @Override
   public native ReadableType getType(int index);
+
+  public List<Object> toJavaArray() {
+    return ReadableNativeArray.toJavaArray(this);
+  }
+
+  public static List<Object> toJavaArray(ReadableNativeArray readableArray) {
+    List<Object> unwrappedList = new ArrayList<>(readableArray.size());
```

Comment:
Why not `ArrayList<Object> arrayList = new ArrayList<>;` and `arrayList.add(value)` instead of `arrayList.put(i, value)`? Since you're looping over the array, the index is the same, and you don't need to specify the size while initialising and the index every time.